src/hooks/useCartUpdateApi.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useRef } from 'react';
+
+const useCartUpdateApi = (productId: number, value: number, setValue: (value: number) => void) => {
+  const prevValue = useRef(value);
+
+  const resetValueOnError = (promise: Promise<Response>, properResStatus: number) => {
+    promise
+      .then((response) => {
+        if (response.status !== properResStatus) setValue(prevValue.current);
+        else prevValue.current = value;
+      })
+      .catch(() => {
+        setValue(prevValue.current);
+      });
+  };
+
+  useEffect(() => {
+    if (prevValue.current === value) return;
+
+    if (prevValue.current === 0 && value > 0) {
+      fetch('/cart-items', { method: 'POST', body: JSON.stringify({ productId }) });
+      resetValueOnError(
+        fetch('/cart-items', { method: 'POST', body: JSON.stringify({ productId }) }),
+        201
+      );
```

Comment:
동일한 API 요청을 두번하고 있네요~ 한번 요청으로 해결하는 방법으로 수정하면 좋을거 같아요

Reply:
```suggestion
      resetValueOnError(
        fetch('/cart-items', { method: 'POST', body: JSON.stringify({ productId }) }),
        201
      );
```
앗 원래 이 부분만 있어도 되는데 실수했네요😥